src/main/java/com/walmartlabs/x12/X12Parser.java

```diff
@@ -53,9 +64,28 @@
             return segments;
         } else {
             // if there is only one line in the source data
-            // we should attempt to use the segment separator passed in
-            // and see if we can split up this source data
-            return splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            // we should attempt to use the default segment separator
+            segments = splitSourceDataIntoSegments(sourceData, "\\" + DEFAULT_SEGMENT_SEPARATOR);
+            if (segments != null && segments.size() > 1) {
+                return segments;
+            } else {
+                // if there is only one line in the source data
+                // now try to get the specified delimiter
+                return splitSourceDataIntoSegments(sourceData, "\\" + findSegmentDelimiterCharacter(sourceData));
+            }
+        }
+    }
+    
+    /**
+     * get the segment delimiter/separator character
+     * @param sourceData
+     * @return the character at the 106th position
+     */
+    default Character findSegmentDelimiterCharacter(String sourceData) {
+        if (sourceData != null && sourceData.length() > SEGMENT_SEPARATOR_INDEX) {
+            return Character.valueOf(sourceData.charAt(SEGMENT_SEPARATOR_INDEX));
+        } else {
+            return null;
```

Comment:
What happens if this method returns null when you call it from here:
`splitSourceDataIntoSegments(sourceData, "\\" + findSegmentDelimiterCharacter(sourceData));`

Reply:
throws `PatternSyntaxException`, I added that to JavaDocs & added a test
good catch